src/functionality.py

```diff
@@ -49,24 +150,652 @@ def _validate_command(ctx: commands.Context) -> bool:
+                                                  f'**Reason**: {user["reason"]}\n'
+                                                  f'**Ban Ended**: {user["ended"]}',
+
+
+                                      colour=discord.Color.dark_blue())
+                embed.set_thumbnail(url=ctx.author.avatar_url)
+                embed.set_footer(text=ctx.author)
+
+                if _all is True:
+                    await ctx.send(embed=embed)
+
+            if _all is False:
+                await ctx.send(embed=embed)
+
+            return
+
+        @bot.command(
+            name='ban_list_update',
+            brief='Forces the ban_list to update.',
+            help=('Forces the ban_list to update. You do not need any arguments to proceed.'
+                  '\nArguments: $ban_list_update'
+                  '\nExample: $ban_list_update'),
+        )
+        async def ban_list_update(ctx: commands.Context):
+            if not _validate_command(ctx):
+                return
+
+            if not admin_check(ctx):
+                return await ctx.send("`Insufficient Privileges.`")
+
+            updated_list = update_ban_list()[1]
+
+            if updated_list:
+                for user in updated_list:
+                    answer = int(user["ban_timestamp"]) + int(user["ban_length"])
+                    answer = round(time.time()) - answer
+
+                    if answer >= 0:
+                        target = await bot.fetch_user(int(user['discord_id']))
+                        await target.send(f'**You are now unbanned from accessing the Server Bot. Please do not make the same offense again.**')
+
+            await ctx.send("**Updated! Those whose ban is over will be notified.**")
+
+        @bot.command(
+            name='add_roleplay',
+            brief='Adds a roleplay into the Database.',
+            help=('Adds a roleplay into the Database. There are a lot of arguments needed to be filled; ensure you do "open and close" quotes IF they are not integers.'
+                  '\nArguments: $add_roleplay <rp_name> <main_host_id> <rp_start_date> <rp_duration> <doc> '
+                  '<serial_code: optional> <local: optional> <sign_up: optional> <ongoing: optional> <ended: optional>'
+                  '\nExample: $add_roleplay "Helvetica Neue" 332456386946531328 42069 5 "https://www.epochconverter.com"'
+                  '"HN" 1 0 0'
+                  "\n\n<main_host_id> is the Main Host's Discord ID."
+                  "\n<rp_start_date> is an epoch number. Use https://www.epochconverter.com"
+                  "\n<rp_duration> is an integer. In hours, how long the Roleplay is."
+                  "\n<serial_code> is an optional field, but can be customized. It is used to find your RP in [rp_profile] and store in the database."
+                  "\n<local> is an optional field, can only be 0 or 1. --0: False (Hosted outside DRO) | 1: True (Hosted within DRO)--"
```

Comment:
"Within DRO" is not semantically valid, because all RPs take place in DRO. You probably meant ODROS?

Reply:
Yeah, I meant ODROS. Wording changed, now.